changelogs/unreleased/fix-bugs-in-merge-partial-compile.yml

```diff
@@ -0,0 +1,4 @@
+---
+description: Fix bugs in the merge logic of a partial compile. 1) Ensure that the version numbers present in the new version of the configuration model are set correctly. 2) Ensure that the resource states and unknowns, that belongs to the partial model, are sent to the server and merged correctly with the old configuration model.
+change-type: patch
```

Comment:
I didn't add one because it was released as being experimental, so you are argue whether this is a bugfix or just ongoing development work. But I don't have a strong opinion on this. I will add it.

Reply:
I thought we had only marked the interface as experimental but indeed you're right that we marked the whole thing as such. So either is fine for me.